pkg/util/ovs/fake_ovs.go

```diff
@@ -0,0 +1,447 @@
+			for _, existingPort := range fake.ports {
+				if existingPort >= ofport {
+					ofport = existingPort + 1
+				}
+			}
+		} else {
+			ofport = ofportRequest
+		}
+		fake.ports[port] = ofport
+	}
+	return ofport, nil
+}
+
+func (fake *ovsFake) DeletePort(port string) error {
+	if err := fake.ensureExists(); err != nil {
+		return err
+	}
+
+	delete(fake.ports, port)
+	return nil
+}
+
+func (fake *ovsFake) SetFrags(mode string) error {
+	return nil
+}
+
+func (ovsif *ovsFake) Create(table string, values ...string) (string, error) {
+	return "fake-UUID", nil
+}
+
+func (fake *ovsFake) Destroy(table, record string) error {
+	return nil
+}
+
+func (fake *ovsFake) Get(table, record, column string) (string, error) {
+	return "", nil
+}
+
+func (fake *ovsFake) Set(table, record string, values ...string) error {
+	return nil
+}
+
+func (fake *ovsFake) Clear(table, record string, columns ...string) error {
+	return nil
+}
+
+type ovsFakeTx struct {
+	fake *ovsFake
+	err  error
+}
+
+func (fake *ovsFake) NewTransaction() Transaction {
+	return &ovsFakeTx{fake: fake, err: fake.ensureExists()}
+}
+
+type parseCmd string
+
+const (
+	parseForAdd    parseCmd = "add-flow"
+	parseForDelete parseCmd = "del-flows"
+)
+
+func fieldSet(parsed *ovsFlow, field string) bool {
+	for _, f := range parsed.fields {
+		if f.name == field {
+			return true
+		}
+	}
+	return false
+}
+
+func checkNotAllowedField(flow string, parsed *ovsFlow, field string, cmd parseCmd) error {
+	if fieldSet(parsed, field) {
+		return fmt.Errorf("bad flow %q (field '%s' not allowed in %s)", flow, field, cmd)
+	}
+	return nil
+}
+
+func checkUnsupportedField(flow string, parsed *ovsFlow, field string) error {
+	if fieldSet(parsed, field) {
+		return fmt.Errorf("bad flow %q (field '%s' not supported)", flow, field)
+	}
+	return nil
+}
+
+func parseFlow(cmd parseCmd, flow string, args ...interface{}) (*ovsFlow, error) {
+	if len(args) > 0 {
+		flow = fmt.Sprintf(flow, args...)
+	}
+
+	// According to the man page, "flow descriptions comprise a series of field=value
+	// assignments, separated by commas or white space." However, you can also have
+	// fields with no value (eg, "ip"), and the "actions" field can also have internal
+	// commas, whitespace, and equals signs (but if it is present, it must be the
+	// last field specified).
+
+	parsed := &ovsFlow{
+		table:    0,
+		priority: defaultPriority,
+		fields:   make([]ovsField, 0),
+		created:  time.Now(),
+		cookie:   "0",
+	}
+	for flow[0] == ' ' {
+		flow = flow[1:]
+	}
+	origFlow := flow
+	for flow != "" {
+		field := ""
+		value := ""
+
+		end := strings.IndexAny(flow, ", ")
+		if end == -1 {
+			end = len(flow)
+		}
+		eq := strings.Index(flow, "=")
+		if eq == -1 || eq > end {
+			field = flow[:end]
+		} else {
+			field = flow[:eq]
+			if field == "actions" {
+				end = len(flow)
+				value = flow[eq+1:]
+			} else {
+				valueEnd := end - 1
+				for flow[valueEnd] == ' ' || flow[valueEnd] == ',' {
+					valueEnd--
+				}
```

Comment:
There could be multiple spaces or comma+spaces in the end
value = strings.TrimRight(flow[eq+1:end], ", ")